From 1cb7442b2ecdf17159eb79e194b9f07bd60d0480 Mon Sep 17 00:00:00 2001
From: Shuzhen Wang <shuzhenwang@google.com>
Date: Fri, 8 Dec 2017 12:16:49 -0800
Subject: [PATCH 03/18] OutputConfiguration: Fix missing mIsShared in parcel
 read

Test: Camera CTS
Bug: 69683251

(cherry picked from commit 4304a02ac990c1af5fb8f479bdd2b04c8af4fddb)
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63420
Signed-off-by: johnzhax <johnx.zhang@intel.com>
Signed-off-by: Walter Yan < walterx.yan@intel.com>

Change-Id: Ib8684c6698fef4fef2c3ddff9c1e7b5e0841a595
---
 core/java/android/hardware/camera2/params/OutputConfiguration.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/java/android/hardware/camera2/params/OutputConfiguration.java b/core/java/android/hardware/camera2/params/OutputConfiguration.java
index 2b317d6..05c4dc3 100644
--- a/core/java/android/hardware/camera2/params/OutputConfiguration.java
+++ b/core/java/android/hardware/camera2/params/OutputConfiguration.java
@@ -409,6 +409,7 @@ public final class OutputConfiguration implements Parcelable {
         this.mConfiguredSize = other.mConfiguredSize;
         this.mConfiguredGenerationId = other.mConfiguredGenerationId;
         this.mIsDeferredConfig = other.mIsDeferredConfig;
+        this.mIsShared = other.mIsShared;
     }
 
     /**
@@ -421,6 +422,7 @@ public final class OutputConfiguration implements Parcelable {
         int width = source.readInt();
         int height = source.readInt();
         boolean isDeferred = source.readInt() == 1;
+        boolean isShared = source.readInt() == 1;
         ArrayList<Surface> surfaces = new ArrayList<Surface>();
         source.readTypedList(surfaces, Surface.CREATOR);
 
@@ -431,6 +433,7 @@ public final class OutputConfiguration implements Parcelable {
         mSurfaces = surfaces;
         mConfiguredSize = new Size(width, height);
         mIsDeferredConfig = isDeferred;
+        mIsShared = isShared;
         mSurfaces = surfaces;
         if (mSurfaces.size() > 0) {
             mSurfaceType = SURFACE_TYPE_UNKNOWN;
-- 
1.9.1

